addons/web/static/tests/search/search_bar_tests.js

```diff
@@ -898,4 +898,69 @@ QUnit.module("Search", (hooks) => {
         await triggerEvent(searchInput, null, "keydown", { key: "ArrowUp" });
         assert.containsOnce(target, ".focus");
     });
+
+    QUnit.test("many2one_reference fields are supported in search view", async function (assert) {
+        const partnerModel = serverData.models.partner;
+        partnerModel.fields.res_id = {
+            string: "Resource ID",
+            type: "many2one_reference",
+            model_name_ref_fname: "res_model",
+            inverse_fname_by_model_name: { bar: "one2many_field" },
+        };
+        partnerModel.fields.res_model = { string: "Resource Model", type: "char" };
+        serverData.models.bar = {
+            fields: {
+                one2many_field: {
+                    type: "one2many",
+                    relation: "partner",
+                },
+            },
+            records: [{ id: 1, display_name: "The only record" }],
+        };
+        partnerModel.records.forEach((record) => {
+            record.res_model = "bar";
+            record.res_id = 1;
+        });
+
+        const controlPanel = await makeWithSearch({
+            serverData,
+            resModel: "partner",
+            Component: ControlPanel,
+            searchMenuTypes: [],
+            searchViewId: false,
+            searchViewArch: /*xml*/ `
+                    <search>
+                        <field name="foo" />
+                        <field name="res_id" />
+                    </search>
+                `,
+        });
+
+        assert.deepEqual(getDomain(controlPanel), []);
+
+        await editSearch(target, "12");
+        assert.containsN(
+            target,
+            ".o_searchview ul li.dropdown-item",
+            2,
+            "should display 2 dropdown menus -- for foo and res_id"
```

Comment:
we tend to put no messages in assert
--> the best think is to add some assert that makes seems clear
(too often, messages have become misleading. an assert cannot)